app/models/metric/ci_mixin/capture.rb

```diff
@@ -32,27 +46,22 @@ def perf_capture_queue(interval_name, options = {})
 
     log_target = "#{self.class.name} name: [#{name}], id: [#{id}]"
 
-    # Determine what items we should be queuing up
-    items = []
     cb = nil
     if interval_name == 'historical'
       start_time = Metric::Capture.historical_start_time if start_time.nil?
       end_time = Time.now.utc if end_time.nil?
-
-      start_hour = start_time
-      while start_hour != end_time
-        end_hour = start_hour + 1.day
-        end_hour = end_time if end_hour > end_time
-        items.unshift([interval_name, start_hour, end_hour])
-        start_hour = end_hour
-      end
     else
-      items << [interval_name]
-      items[0] << start_time << end_time unless start_time.nil?
-
+      start_time = last_perf_capture_on unless start_time
```

Comment:
> One other point to note is that for historical collection, we intentionally collected the hours in reverse (that's the .unshift in the original). This was because historical is done in a lower priority. So, it's possible data could come in more slowly and we wanted the charts to grow backwards instead of growing forward but from a time in the way past (if that makes sense). This code undoes that.

D'oh, I'll go over that one more time, then.  I think I can just reverse what I've got now with the steps and each_cons bits.